### PR TITLE
[16.0][IMP] stock_account: refactoring for checking restrict_partner_id

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
Refactored the restrict_parnter_id check in the stock_account module to make it more flexible for future updates.The The purpose of this refactoring is to prepare for a custom module that will value inventory based on specific ownership.
Related: https://github.com/OCA/stock-logistics-workflow/pull/1548

@qrtl QT4334

### CODE REVIEW

**Comment by AungKoKoLin1997** [comment]
@clesgow @pfertyk Can you please review this PR? I create this PR as a follow-up that was missed in this PR https://github.com/odoo/odoo/pull/155694.

**Comment by pfertyk** [comment]
Hi @AungKoKoLin1997 ! Could you please fix the trailing whitespace issue reported by the style check? :)

**Comment by AungKoKoLin1997** [comment]
@pfertyk Fixed it!

File: addons/stock_account/models/stock_move.py [hunk 1]
```diff
@@ -555,15 +555,23 @@ def _account_analytic_entry_move(self):
             self.env['stock.move'].browse(
                 move_id).analytic_account_line_id = analytic_line
 
+    @api.model
```
Comment (clesgow) [inline addons/stock_account/models/stock_move.py:558]:
```suggestion
```
As the method uses the content of `self`, `@api.model` doesn't apply here. More info in our [documentation](https://www.odoo.com/documentation/16.0/developer/reference/backend/orm.html#odoo.api.model).

Comment (AungKoKoLin1997) [reply addons/stock_account/models/stock_move.py:558]: I see. Thanks for the pointer.

**Review by clesgow — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**